src/pybind/mgr/cephadm/schedule.py

```diff
@@ -362,6 +363,11 @@ def expand_candidates(ls: List[DaemonPlacement], num: int) -> List[DaemonPlaceme
     def find_ip_on_host(self, hostname: str, subnets: List[str]) -> Optional[str]:
         for subnet in subnets:
             ips: List[str] = []
+            # following is to allow loopback interfaces for both ipv4 and ipv6
+            if ipaddress.ip_address('127.0.0.1') in ipaddress.ip_network(subnet):
```

Comment:
interesting, I didn't know you could just do a `<ip> in <network>` check to see if the ip was in a certain subnet. We could maybe use similar to address https://tracker.ceph.com/issues/57060#note-7

Reply:
> interesting, I didn't know you could just do a `<ip> in <network>` check to see if the ip was in a certain subnet. We could maybe use similar to address https://tracker.ceph.com/issues/57060#note-7

Yeah, it's very handy (and clean) to perform the check. The code behind it is the following. I think we can use it to solve the issue you referenced :+1: 

```
    def __contains__(self, other):
        # always false if one is v4 and the other is v6.
        if self._version != other._version:
            return False
        # dealing with another network.
        if isinstance(other, _BaseNetwork):
            return False
        # dealing with another address
        else:
            # address
            return other._ip & self.netmask._ip == self.network_address._ip

```